### PR TITLE
Tracks fragment added

### PR DESCRIPTION
**Pull request by mananwason**
@creativepsyco please check and merge.

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            itemView.setClickable(true);
+            itemView.setOnClickListener(this);
+            title = (TextView) itemView.findViewById(R.id.track_title);
+            desc = (TextView) itemView.findViewById(R.id.track_description);
+        }
+
+
+        @Override
+        public void onClick(View view) {
+            Toast.makeText(context, "Items   :" + getItemCount(), Toast.LENGTH_SHORT).show();
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java]: Let's not do this?

File: app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by MananWason on 07-06-2015.
+ */
+public class TracksListAdapter extends RecyclerView.Adapter<TracksListAdapter.Viewholder> {
+
+    List<Track> tracks ;
+    private Context context;
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java]: Don't need to cache the context or the inflater.

File: app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    public TracksListAdapter(Context context, List<Track> tracks) {
+        this.context = context;
+        layoutInflater = LayoutInflater.from(context);
+        this.tracks = tracks;
+    }
+
+    @Override
+    public TracksListAdapter.Viewholder onCreateViewHolder(ViewGroup parent, int viewType) {
+
+        View view = layoutInflater.inflate(R.layout.tracks_item, parent, false);
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java]: We can use `LayoutInflator.from(parent.getContext())...`

File: app/src/main/java/org/fossasia/openevent/data/Track.java [hunk 4]
```diff
@@ -31,9 +33,14 @@ public String getDescription() {
...
     public String generateSql() {
-        String query_normal = "INSERT INTO %s VALUES ('%d', '%s', '%s');";
-        String query = String.format(query_normal, DbContract.Tracks.TABLE_NAME, id, name, description);
+        String query_normal = "INSERT INTO %s VALUES ('%d', '%s', %s);";
+        String query = String.format(query_normal, DbContract.Tracks.TABLE_NAME, id, name, DatabaseUtils.sqlEscapeString(description));
         return query;
 
     }
+
+    private String escapeChar(String string) {
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/data/Track.java]: You should probably make this into a utility function.

File: app/src/main/java/org/fossasia/openevent/data/Version.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Created by MananWason on 05-06-2015.
+ */
+public class Version {
+    @SerializedName("event_id")
+    int eventId;
+    @SerializedName("event_ver")
+    int eventVer;
+    int id;
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/data/Version.java]: what specific use case we have for version having an id field

File: app/src/main/java/org/fossasia/openevent/fragments/SponsorsFragment.java [hunk 6]
```diff
@@ -1,53 +1,21 @@
...
-    }
+public class SponsorsFragment extends ListFragment {
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
 
-        }
+        DbSingleton dbSingleton = DbSingleton.getInstance();
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/fragments/SponsorsFragment.java]: requires cleanup, No use happening here.

File: app/src/main/res/layout/tracks_item.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+              android:layout_width="match_parent"
+              android:layout_height="match_parent"
+              android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/track_title"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textColor="#000"
```
Comment (creativepsyco) [inline app/src/main/res/layout/tracks_item.xml]: 6 color hex codes please.

**Comment by creativepsyco** [comment]
Some cleanup still required. After that we can do the merge.

**Comment by mananwason** [comment]
@creativepsyco This commit was somewhere between the completed tracks and the sponsors, so either I can commit the sponsors fragment and then you can merge both of them in the same PR or maybe you merge this and I'll make the changes in the next commit. Please tell me how should we proceed?

**Comment by creativepsyco** [comment]
merged

File: app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java (hunk 2 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java]: So how do you suggest I should take the context?

File: app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java (hunk 3 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/Adapters/TracksListAdapter.java]: I can use this in oncreateViewHolder but what about in onbindviewholder?

File: app/src/main/java/org/fossasia/openevent/data/Version.java (hunk 5 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/data/Version.java]: @creativepsyco The version id will be the primary check. If version id is changed then there is same change in the data.

**Pull request merged**